### PR TITLE
Add NextAlone's Utils, add entry for AutoRenewFire, simplify chat long click menu

### PR DESCRIPTION
**Pull request by NextAlone**
# Add NextAlone's Utils, add entry for AutoRenewFire, simplify chat long click menu

## 介绍 / Description

- Fix typo …
- Add NextAlone's utils and modify LogUtils …
- Add entry for AutoRenewFire …
- Simplify chat long click menu …

## 更改内容 / Types of Changes

- 修复 Bugfix
- 新功能 New Feature

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [ ] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/nil/nadph/qnotified/activity/SettingsActivity.java [hunk 1]
```diff
@@ -76,8 +102,27 @@
 import static me.singleneuron.util.KotlinUtilsKt.addViewConditionally;
 import static me.singleneuron.util.QQVersion.QQ_8_2_0;
 import static me.singleneuron.util.QQVersion.QQ_8_2_6;
-import static nil.nadph.qnotified.ui.ViewBuilder.*;
```
Comment (singleNeuron) [inline app/src/main/java/nil/nadph/qnotified/activity/SettingsActivity.java:79]: 别天天import打仗了求求你们了

**Review by singleNeuron — CHANGES_REQUESTED** [review]

File: app/src/main/java/nil/nadph/qnotified/activity/AlphaTestFuncActivity.java [hunk 2]
```diff
@@ -85,20 +77,27 @@ public boolean doOnCreate(Bundle bundle) {
             View v = subtitle(this, "狐狸狸测试功能");
             v.setOnClickListener(v1 -> RevokeMsgDialog.INSTANCE.onShow(AlphaTestFuncActivity.this));
             ll.addView(v);
+            View autoRenewFire = newListItemButton(this, "[特供版]自动续火", "芜湖", null,
+                view -> AutoRenewFireDialog.INSTANCE.showMainDialog(this));
+            autoRenewFire.setOnLongClickListener(view -> {
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/AlphaTestFuncActivity.java:82]: 你塞在里边用户用不了的

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 3]
```diff
@@ -54,128 +72,115 @@
...
+    
     private int myId = -1;
-
+    
     @NonNull
     public static AbsDelayableHook getHookByType(int hookId) {
         return queryDelayableHooks()[hookId];
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:88]: ？

Comment (cinit) [reply app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:88]: ?

**Review by cinit — CHANGES_REQUESTED** [review]

**Review by cinit — DISMISSED** [review]

File: app/src/main/java/nil/nadph/qnotified/activity/AlphaTestFuncActivity.java [hunk 4]
```diff
@@ -18,34 +18,26 @@
...
 
-import android.annotation.SuppressLint;
-import android.os.Bundle;
-import android.view.View;
-import android.view.ViewGroup;
-import android.view.WindowManager;
-import android.widget.LinearLayout;
-import android.widget.RelativeLayout;
-import android.widget.TextView;
+import android.annotation.*;
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/AlphaTestFuncActivity.java:21]: 按需import尽量别*

File: app/src/main/java/nil/nadph/qnotified/activity/AlphaTestFuncActivity.java [hunk 5]
```diff
@@ -85,20 +77,27 @@ public boolean doOnCreate(Bundle bundle) {
             View v = subtitle(this, "狐狸狸测试功能");
             v.setOnClickListener(v1 -> RevokeMsgDialog.INSTANCE.onShow(AlphaTestFuncActivity.this));
             ll.addView(v);
+            View autoRenewFire = newListItemButton(this, "[特供版]自动续火", "芜湖", null,
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/AlphaTestFuncActivity.java:80]: 塞这里边用户用不了

File: app/src/main/java/nil/nadph/qnotified/activity/SettingsActivity.java [hunk 6]
```diff
@@ -18,64 +18,43 @@
...
-import nil.nadph.qnotified.util.Initiator;
-import nil.nadph.qnotified.ui.CustomDialog;
-import nil.nadph.qnotified.ui.HighContrastBorder;
-import nil.nadph.qnotified.ui.ResUtils;
-import nil.nadph.qnotified.util.LicenseStatus;
-import nil.nadph.qnotified.util.NewsHelper;
-import nil.nadph.qnotified.util.Toasts;
-import nil.nadph.qnotified.util.UpdateCheck;
-import nil.nadph.qnotified.util.Utils;
+import nil.nadph.qnotified.ui.*;
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/SettingsActivity.java:51]: 按需import

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 7]
```diff
@@ -54,128 +72,115 @@
...
-import com.rymmmmm.hook.OneTapTwentyLikes;
-import com.rymmmmm.hook.RemoveMiniProgramAd;
-import com.rymmmmm.hook.RemoveSendGiftAd;
-import com.rymmmmm.hook.ShowMsgCount;
 import nil.nadph.qnotified.step.Step;
 import nil.nadph.qnotified.util.Utils;
 
 public abstract class AbsDelayableHook implements SwitchConfigItem {
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:79]: 迷惑行为

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 8]
```diff
@@ -54,128 +72,115 @@
...
-import com.rymmmmm.hook.ShowMsgCount;
 import nil.nadph.qnotified.step.Step;
 import nil.nadph.qnotified.util.Utils;
 
 public abstract class AbsDelayableHook implements SwitchConfigItem {
-
+    
     private static AbsDelayableHook[] sAllHooks;
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:81]: same

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 9]
```diff
@@ -54,128 +72,115 @@
...
 
 public abstract class AbsDelayableHook implements SwitchConfigItem {
-
+    
     private static AbsDelayableHook[] sAllHooks;
-
+    
     private int myId = -1;
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:83]: same

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 10]
```diff
@@ -188,26 +193,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
             Utils.log(e);
         }
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:196]: same

**Review by qwq233 — CHANGES_REQUESTED** [review]

File: app/src/main/java/me/kyuubiran/dialog/AutoRenewFireDialog.kt [hunk 11]
```diff
@@ -38,9 +38,9 @@ import nil.nadph.qnotified.util.LicenseStatus
 object AutoRenewFireDialog {
     private var currentEnable: Boolean? = null
     private lateinit var binding: KyuubiranAutoRenewFireBinding
-    var replyMsg: String = getExFriendCfg().getStringOrDefault(AutoRenewFireMgr.MESSAGE, "火")
+    var replyMsg: String = getExFriendCfg().getStringOrDefault(AutoRenewFireMgr.MESSAGE, "🔥")
```
Comment (qwq233) [inline app/src/main/java/me/kyuubiran/dialog/AutoRenewFireDialog.kt:41]:
```suggestion
    var replyMsg: String = getExFriendCfg().getStringOrDefault(AutoRenewFireMgr.MESSAGE, "[续火]")
```

File: app/src/main/java/nil/nadph/qnotified/activity/OmegaTestFuncActivity.java [hunk 12]
```diff
@@ -89,25 +86,33 @@ public void run() {
...
             ll.addView(subtitle(this, "想要隐私就不要去玩 群发 或者 卡片消息, 是否开启功能是你们的自由", Color.RED));
             ll.addView(subtitle(this, "如果您看不懂, 或无法理解以上内容, 请勿使用 群发 或 卡片消息 功能!", Color.RED));
         }
         __ll.setLayoutParams(new ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT));
-        this.setContentView(bounceScrollView);
+        setContentView(bounceScrollView);
         LinearLayout.LayoutParams _lp_fat = new LinearLayout.LayoutParams(MATCH_PARENT, 0);
         _lp_fat.weight = 1;
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/OmegaTestFuncActivity.java:111]:
```suggestion
```

File: app/src/main/java/nil/nadph/qnotified/activity/OmegaTestFuncActivity.java [hunk 13]
```diff
@@ -89,25 +86,33 @@ public void run() {
...
         _lp_fat.weight = 1;
-
+        
         setContentBackgroundDrawable(ResUtils.skin_background);
         setTitle("Omega测试性功能");
         getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
         return true;
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/OmegaTestFuncActivity.java:117]:
```suggestion
```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 14]
```diff
@@ -20,164 +20,126 @@
...
-import com.rymmmmm.hook.DisablePokeEffect;
-import com.rymmmmm.hook.IgnoreDiyCard;
-import com.rymmmmm.hook.OneTapTwentyLikes;
-import com.rymmmmm.hook.RemoveMiniProgramAd;
-import com.rymmmmm.hook.RemoveSendGiftAd;
-import com.rymmmmm.hook.ShowMsgCount;
-import nil.nadph.qnotified.step.Step;
-import nil.nadph.qnotified.util.Utils;
+import com.rymmmmm.hook.*;
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:24]:
```suggestion
```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 15]
```diff
@@ -20,164 +20,126 @@
...
+import me.singleneuron.hook.*;
+import me.singleneuron.hook.decorator.*;
+import me.singleneuron.qn_kernel.dispacher.*;
+import nil.nadph.qnotified.config.*;
+import nil.nadph.qnotified.step.*;
+import nil.nadph.qnotified.util.*;
 
 public abstract class AbsDelayableHook implements SwitchConfigItem {
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:37]:
```suggestion
```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 16]
```diff
@@ -20,164 +20,126 @@
...
+import nil.nadph.qnotified.config.*;
+import nil.nadph.qnotified.step.*;
+import nil.nadph.qnotified.util.*;
 
 public abstract class AbsDelayableHook implements SwitchConfigItem {
-
+    
     private static AbsDelayableHook[] sAllHooks;
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:39]:
```suggestion
```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 17]
```diff
@@ -20,164 +20,126 @@
...
 
 public abstract class AbsDelayableHook implements SwitchConfigItem {
-
+    
     private static AbsDelayableHook[] sAllHooks;
-
+    
     private int myId = -1;
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:41]:
```suggestion
```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 18]
```diff
@@ -20,164 +20,126 @@
...
+    
     private int myId = -1;
-
+    
     @NonNull
     public static AbsDelayableHook getHookByType(int hookId) {
         return queryDelayableHooks()[hookId];
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:46]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 19]
```diff
@@ -20,164 +20,126 @@
...
+                SortAtPanel.INSTANCE,
+                SendFavoriteHook.INSTANCE,
+                AutoRenewFire.INSTANCE,
+                SimplifyChatLongItem.INSTANCE,
             };
         }
         return sAllHooks;
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:142]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 20]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
             Utils.log(e);
         }
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:155]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 21]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
             Utils.log(e);
         }
     }
-
+    
     public abstract boolean isTargetProc();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:157]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 22]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
         }
     }
-
+    
     public abstract boolean isTargetProc();
-
+    
     public abstract int getEffectiveProc();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:159]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 23]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
+    
     public abstract boolean isTargetProc();
-
+    
     public abstract int getEffectiveProc();
-
+    
     public abstract boolean isInited();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:161]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 24]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
+    
     public abstract int getEffectiveProc();
-
+    
     public abstract boolean isInited();
-
+    
     public abstract boolean init();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:163]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 25]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
     public abstract boolean isInited();
-
+    
     public abstract boolean init();
-
+    
     @Override
     public abstract boolean sync();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:166]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 26]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
-
+    
     @Override
     public abstract boolean sync();
-
+    
     @NonNull
     public abstract Step[] getPreconditions();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:169]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 27]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
-
+    
     @NonNull
     public abstract Step[] getPreconditions();
-
+    
     @Override
     public abstract boolean isValid();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:172]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 28]
```diff
@@ -190,26 +152,26 @@ public static void allowEarlyInit(@NonNull AbsDelayableHook hook) {
...
     public abstract Step[] getPreconditions();
-
+    
     @Override
     public abstract boolean isValid();
-
+    
     public abstract boolean checkPreconditions();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:174]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 29]
```diff
@@ -223,21 +185,21 @@ public final int getId() {
         }
         return -1;
     }
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:188]:
```suggestion

```

File: app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java [hunk 30]
```diff
@@ -223,21 +185,21 @@ public final int getId() {
...
+    
     /**
      * safe to call, no Throwable allowed
      *
      * @return whether the config item is enabled
      */
     @Override
     public abstract boolean isEnabled();
-
+
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/hook/AbsDelayableHook.java:196]:
```suggestion

```

**Review by qwq233 — APPROVED** [review]

**Review by singleNeuron — APPROVED** [review]

**Review by ketal178 — APPROVED** [review]

File: app/src/main/java/me/nextalone/util/LogUtils.kt [hunk 31]
```diff
@@ -30,14 +29,32 @@ fun logd(msg: String) {
...
+fun logds(msg: String = "") {
+    logd("Start--$msg")
+}
+
+fun logdb(msg: String = "") {
+    logd("Before--$msg")
+}
+
+fun logda(msg: String = "") {
+    logd("After--$msg")
```
[the remark this answers is not in the document]
Comment (NextAlone) [reply app/src/main/java/me/nextalone/util/LogUtils.kt:57]: 好耶

**Pull request merged**